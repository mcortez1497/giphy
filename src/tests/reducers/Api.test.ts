import {
  apiRequestClear,
  apiRequestError,
  apiRequestStart,
  apiRequestSuccess,
  apiReducer
} from "reducers";

const initialState = {};

describe("API State", () => {
  describe("Actions", () => {
    it("returns apiRequestClear action", () => {
      expect(apiRequestClear("test")).toMatchSnapshot();
    });

    it("returns apiRequestError action", () => {
      expect(apiRequestError("test", "error")).toMatchSnapshot();
    });

    it("returns apiRequestStart action", () => {
      expect(apiRequestStart("test")).toMatchSnapshot();
    });

    it("returns apiRequestSuccess action", () => {
      expect(apiRequestSuccess("test")).toMatchSnapshot();
    });
  });

  describe("Reducer", () => {
    it("resets API key/value", () => {
      const action = {
        type: "API_REQUEST_CLEAR",
        request: "test"
      };
      expect(apiReducer(initialState, action)).toMatchSnapshot();
    });

    it("sets API key/value as errored", () => {
      const action = {
        type: "API_REQUEST_ERROR",
        request: "test",
        message: "error"
      };
      expect(apiReducer(initialState, action)).toMatchSnapshot();
    });

    it("sets API key/value as loading", () => {
      const action = {
        type: "API_REQUEST_START",
        request: "test"
      };
      expect(apiReducer(initialState, action)).toMatchSnapshot();
    });

    it("sets API key/value as success", () => {
      const action = {
        type: "API_REQUEST_SUCCESS",
        request: "test"
      };
      expect(apiReducer(initialState, action)).toMatchSnapshot();
    });
  });
});
