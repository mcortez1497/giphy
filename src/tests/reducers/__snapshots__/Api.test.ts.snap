// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API State Actions returns apiRequestClear action 1`] = `
Object {
  "request": "test",
  "type": "API_REQUEST_CLEAR",
}
`;

exports[`API State Actions returns apiRequestError action 1`] = `
Object {
  "message": "error",
  "request": "test",
  "type": "API_REQUEST_ERROR",
}
`;

exports[`API State Actions returns apiRequestStart action 1`] = `
Object {
  "request": "test",
  "type": "API_REQUEST_START",
}
`;

exports[`API State Actions returns apiRequestSuccess action 1`] = `
Object {
  "request": "test",
  "type": "API_REQUEST_SUCCESS",
}
`;

exports[`API State Reducer resets API key/value 1`] = `
Object {
  "test": Object {
    "isError": false,
    "isLoading": false,
    "isSuccess": false,
  },
}
`;

exports[`API State Reducer sets API key/value as errored 1`] = `
Object {
  "test": Object {
    "isError": true,
    "isLoading": false,
    "isSuccess": false,
    "message": "error",
  },
}
`;

exports[`API State Reducer sets API key/value as loading 1`] = `
Object {
  "test": Object {
    "isError": false,
    "isLoading": true,
    "isSuccess": false,
  },
}
`;

exports[`API State Reducer sets API key/value as success 1`] = `
Object {
  "test": Object {
    "isError": false,
    "isLoading": false,
    "isSuccess": true,
  },
}
`;
